### ISSUE TITLE
Repo manager picks up hidden files

### ISSUE THREAD

**Issue by david4096** [ga4gh/server#945]
Should it fail if there are hidden files in the directories? How pristine do they need to be?

```
top-level directory contents should be '[u'datasets', u'ontologies', u'referenceSets']' but are actually '['.DS_Store', 'datasets', 'g2pDatasets', 'ontologies', 'referenceSets']'
```

**Comment by jeromekelleher** [comment]
This is probably a bug all right, since this '.DS_Store' thing seems to be something that macs like doing, and people seem to like using macs for some reason.

@dcolligan, thoughts?

**Comment by jmarshall** [comment]
IMHO it should just be checking that the directories it needs are there (as it does just after that check).

If it also does this check that there is nothing additional there that this version of the server doesn't know about… this makes it harder to exercise multiple versions of the server against the same data repo tree or to run additional specialised things.

In the given example, having a `g2pDatasets` directory there too seems potentially legit, and not worthy of a complaint.

**Comment by jeromekelleher** [comment]
All sounds sensible to me @jmarshall. We could always include a 'strict' option which insists on only these exact directories existing, or a (default) more relaxed mode which just checks if the directories that it expects exists and it ignores anything else.

**Comment by dcolligan** [comment]
Yeah, I'll rewrite the code to just ensure that the things it needs are there.  Perhaps we can add a 'strict mode' later.

**Comment by david4096** [comment]
Have we documented that the server needs to be bounced when adding data? Not seeing it

**Comment by dcolligan** [comment]
I don't believe so.  Feel free to submit a PR if you think the docs need that.

**Comment by dcolligan** [comment]
Fixed in #948
